### DISCUSSION TITLE
Installation fails if there's an empty file in a subfolder

### DISCUSSION THREAD

**Discussion by luiscantero** [datenstrom/community#674] [Report a bug] [answered]
I dowloaded the latest version from Github, unzipped and uploaded it to my web server. My server works well as it runs Wordpress and Grav just fine.

When I open the yellow page, I get the following error:
`Datenstrom Yellow requires complete upload!`

I changed a line in yellow-system.ini from this:
CoreExtensionDirectory: system/extensions/
... to this:
CoreExtensionDirectory: /system/extensions/
... and the error went away.

The next error I'm getting is:
`Language 'en' does not exist!`

What do I need to do to make it work?

Thanks.

**Comment by markseuffert** [comment]
I would do the installation all over again from scratch, you are probably just missing files. 

Here's what the [troubleshooting](https://datenstrom.se/yellow/help/troubleshooting) recommends in this case: Copy all files to the web server. Check if your FTP software shows an error message during upload. It sometimes happens that the data transfer was interrupted, then there are files with zero bytes on the web server. After all installation files have been copied to the web server, the problem should be resolved.

Does this work for you?

**Reply by luiscantero** [reply]
I redid the installation multiple times. I followed the troubleshooting steps and checked that no files had zero-length.

If you read my posting again you will notice that I actually solved the first issue about complete upload by modifying the file yellow-system.ini, so I'm wondering if that's a bug in the product.

The issue that I'm currently having is about the language not being found, see posting.

**Reply by schulle4u** [reply]
Can you tell a bit more about your hosting environment? Web server, php version etc. 
In most cases Datenstrom Yellow should run without any modifications in `yellow-system.ini`. The second error you came accross probably is caused by your directory modification, resulting in an invalid directory to search for extensions. Directory entries should be relative to Yellow's root directory.

**Reply by luiscantero** [reply]
Here are some lines from phpinfo:
```
PHP Version 7.4.23
System	Linux cpanel-001-******.com 3.10.0-962.3.2.lve1.5.36.el7.x86_64 #1 SMP Mon May 18 02:16:06 EDT 2020 x86_64
Build Date	Aug 24 2021 17:39:25
Configure Command	'./configure' '--disable-dependency-tracking' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/libexec' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' '--target=x86_64-redhat-linux-gnu' '--program-prefix=' '--prefix=/opt/alt/php74' '--exec-prefix=/opt/alt/php74' '--bindir=/opt/alt/php74/usr/bin' '--sbindir=/opt/alt/php74/usr/sbin' '--sysconfdir=/opt/alt/php74/etc' '--datadir=/opt/alt/php74/usr/share' '--includedir=/opt/alt/php74/usr/include' '--libdir=/opt/alt/php74/usr/lib64' '--libexecdir=/opt/alt/php74/usr/libexec' '--localstatedir=/var' '--with-curl=/opt/alt/curlssl11/usr' '--sharedstatedir=/usr/com' '--mandir=/opt/alt/php74/usr/share/man' '--infodir=/opt/alt/php74/usr/share/info' '--cache-file=../config.cache' '--with-libdir=lib64' '--with-config-file-path=/opt/alt/php74/etc' '--with-config-file-scan-dir=/opt/alt/php74/link/conf' '--disable-debug' '--enable-calendar' '--enable-exif' '--enable-ftp' '--enable-huge-code-pages' '--enable-opcache' '--enable-shmop' '--enable-xml' '--with-bz2' '--with-freetype=/usr' '--with-gettext' '--with-gmp' '--with-iconv' '--with-jpeg=/usr' '--with-kerberos' '--with-layout=GNU' '--with-mhash' '--with-openssl-dir=/opt/alt/openssl11' '--with-openssl=/opt/alt/openssl11' '--with-password-argon2=/usr' '--with-pcre-jit' '--with-external-pcre' '--with-pic' '--with-readline' '--with-webp=/usr' '--with-xpm=/usr' '--with-zlib' '--with-zlib-dir=/usr' '--without-gdbm' '--without-pear' '--enable-litespeed' '--enable-pcntl' '--without-mysqli' '--disable-mbstring' '--disable-bcmath' '--disable-dba' '--disable-dom' '--disable-fileinfo' '--disable-json' '--disable-intl' '--disable-pdo' '--disable-phar' '--disable-posix' '--disable-soap' '--disable-sockets' '--disable-sysvsem' '--disable-sysvshm' '--disable-sysvmsg' '--disable-xmlreader' '--disable-xmlwriter' '--without-zip' '--disable-gd' '--without-imap' '--without-xmlrpc' '--without-xsl' '--without-ldap' '--without-pgsql' '--without-snmp' '--without-sodium' '--without-tidy' '--without-enchant' '--without-pspell' '--without-unixODBC' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'target_alias=x86_64-redhat-linux-gnu' 'PKG_CONFIG_PATH=/opt/alt/curlssl11/usr/lib64/pkgconfig:/opt/alt/openssl11/lib64/pkgconfig:/opt/alt/libxml2/usr/lib64/pkgconfig:/opt/alt/sqlite/usr/lib64/pkgconfig:/opt/alt/libicu/usr/lib64/pkgconfig:/opt/alt/pcre2/usr/lib64/pkgconfig:/opt/alt/libgd/lib64/pkgconfig:/opt/alt/libzip/usr/lib64/pkgconfig:/opt/alt/krb5/usr/lib64/pkgconfig::/usr/lib64/pkgconfig:/usr/share/pkgconfig' 'CFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -fno-strict-aliasing -Wno-pointer-sign' 'KERBEROS_CFLAGS=-I/usr/include' 'KERBEROS_LIBS=-L/usr/lib64' 'CXXFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic' 'SASL_CFLAGS=-I/usr/include' 'SASL_LIBS=-L/usr/lib64'
Server API	LiteSpeed V7.9 Cloudlinux 1.3
Virtual Directory Support	disabled
Configuration File (php.ini) Path	/opt/alt/php74/etc
Loaded Configuration File	/opt/alt/php74/etc/php.ini
Scan this dir for additional .ini files	/opt/alt/php74/link/conf
Additional .ini files parsed	/opt/alt/php74/link/conf/alt_php.ini
PHP API	20190902
PHP Extension	20190902
Zend Extension	320190902
Zend Extension Build	API320190902,NTS
PHP Extension Build	API20190902,NTS
Debug Build	no
Thread Safety	disabled
Zend Signal Handling	enabled
Zend Memory Manager	enabled
Zend Multibyte Support	provided by mbstring
IPv6 Support	enabled
DTrace Support	disabled
Registered PHP Streams	https, ftps, compress.zlib, compress.bzip2, php, file, glob, data, http, ftp, phar, zip
Registered Stream Socket Transports	tcp, udp, unix, udg, ssl, sslv3, tls, tlsv1.0, tlsv1.1, tlsv1.2, tlsv1.3
Registered Stream Filters	zlib.*, bzip2.*, convert.iconv.*, string.rot13, string.toupper, string.tolower, string.strip_tags, convert.*, consumed, dechunk
Zend logoThis program makes use of the Zend Scripting Language Engine:
Zend Engine v3.4.0, Copyright (c) Zend Technologies
```

**Reply by IngoBelka** [reply]
I can't confirm the problems. I did right in this moment a fresh install of Yellow and it runs with not an error. Might it possible, that your transfer program does change files while upload? What about if you transfer in binary mode?

**Reply by luiscantero** [reply]
I'm hosting yellow at the root level (not under a subdomain).

**Reply by luiscantero** [reply]
I'm using FileZilla to transfer the files.

**Reply by IngoBelka** [reply]
See [Troubleshooting](https://datenstrom.se/yellow/help/troubleshooting)

Install the missing PHP extension on your web server. You need curl, gd, mbstring and zip.

**Reply by luiscantero** [reply]
@IngoBelka I re-downloaded the files from master, set to binary mode in FileZilla and uploaded the files (after clearing existing ones). Now I get `Datenstrom Yellow requires complete upload!` again.

**Reply by luiscantero** [reply]
@IngoBelka it's all there:
```
curl
cURL support	enabled
cURL Information	7.71.0
gd
GD Support	enabled
GD headers Version	2.2.5
GD library Version	2.2.5
mbstring
Multibyte Support	enabled
Multibyte string engine	libmbfl
zip
Zip	enabled
Zip version	1.15.6
Libzip headers version	1.3.2
Libzip library version	1.3.2
zlib
ZLib Support	enabled
```

**Reply by luiscantero** [reply]
A file yellow.log does not get generated. Setting debug level to 3 does not show me anything obvious. Thanks.

**Reply by IngoBelka** [reply]
You do have root access, don't you? Upload the yellow-master.zip, unpack on your server and copy to the path of your choice. Check if there are file(s) with zero size. If not it should run.

**Reply by IngoBelka** [reply]
> A file yellow.log does not get generated. Setting debug level to 3 does not show me anything obvious. Thanks.

... something in the Apache logs?

**Reply by luiscantero** [reply]
I found the issue by debugging `checkServerComplete`. The method `getDirectoryEntriesRecursive` checks all files in all subfolders recursively, not just the yellow-related files and folders. On my server, there were folders from other CMS systems that I was testing and there was an empty file in one of those.

IMHO the method `checkServerComplete` should only check the yellow files and folders.

Thanks, feel free to close this issue.

**Comment by markseuffert** [comment, marked as answer]
I updated the [install extension](https://github.com/annaesvensson/yellow-install). The new code will check only required files and is completely agnostic to what else is going on in the file system. That should fit better into the rest of the system. There is also a bit more debug information. With debug level 1 you should be able to see which installation files are missing on the web server.

Should be fixed now, please try it out.

**Reply by luiscantero** [reply]
Thanks a lot!
I can confirm that it works even when there are unrelated folders with empty files.
